packages/nocodb/src/lib/meta/api/tableApis.ts

```diff
@@ -277,15 +277,11 @@ export async function tableUpdate(req: Request<any, any>, res) {
     );
   }
 
-  if (
-    !(await Model.checkTitleAvailable({
-      table_name: req.body.table_name,
-      project_id: project.id,
-      base_id: base.id,
-    }))
-  ) {
-    NcError.badRequest('Duplicate table name');
-  }
+  const schould_rename = await Model.checkTitleAvailable({
+    table_name: req.body.table_name,
+    project_id: project.id,
+    base_id: base.id,
+  });
```

Comment:
What if there are tables named `a` and `b` then the user tried to rename `a` as `b`, `should_rename` will be false but it won't return an error. The expected behavior is a bad request error with the `duplicate table name. 


Here you have to get the current table metadata and compare the name.

Reply:
yes you are right thank you
I will adjust that.